### PR TITLE
"// format: off" enabled for comments, literals, etc.

### PR DESCRIPTION
**Pull request by poslegm**
Fixes https://github.com/scalameta/scalafmt/issues/1424

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala [hunk 1]
```diff
@@ -71,6 +72,11 @@ class FormatWriter(formatOps: FormatOps) {
       }
 
       entry.formatWhitespace(sb)
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: just curious: will this introduce any (unwanted) alignment here if `format:off` had been used?

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala [hunk 2]
```diff
@@ -28,12 +27,14 @@ class FormatWriter(formatOps: FormatOps) {
...
     val locations = getFormatLocations(state, debug = false)
 
+    var formatOff = false
     locations.iterate.foreach { entry =>
       val location = entry.curr
       val state = location.state
       val formatToken = location.formatToken
 
       formatToken.left match {
+        case token if formatOff => sb.append(token.syntax)
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: optional:

```
  case token if state.prev.formatOff => sb.append(...)
```

and get rid of `formatOff` variable.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala (hunk 1 above)
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: Good catch, fixed in https://github.com/scalameta/scalafmt/pull/1773/commits/1589507b3df085fa80eb5ac28210eb9fe4307406

Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: Sorry, didn't occur to me before: what about multiple spaces in the original code? Perhaps we should just add the contents of FormatToken.between?

Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: Added test for multiple spaces https://github.com/scalameta/scalafmt/pull/1773/commits/3185a373616ac507bb0fe4254370307b09379493 Or did you mean other cases?

**Pull request merged**